### DISCUSSION TITLE
Realtime API not working with Flutter Web

### DISCUSSION THREAD

**Discussion by Sgambe33** [pocketbase/pocketbase#1485] [Q&A] [answered]
Hi everyone. I've built a web app with flutter and pocketbase. I've implemented the realtime api as the documentation says but when I update the records in the subscribed collection, the values in the web app do not update. After some time I receive this error in the browser console: 
```
{code: 404, message: "Missing or invalid client id.", data: {}}
```
I've read another discussion and found out it is because the web app is not sending the client id to the server. How do I do it?

**Comment by ganigeorgiev** [comment, marked as answer]
The realtime apis are currently not supported in the PocketBase Dart SDK on web platforms because the std http client doesn't support streamed responses out of the box on the web yet - https://github.com/dart-lang/http/issues/595. This and other limitations you could find in https://github.com/pocketbase/dart-sdk/#limitations.

If you are targeting only web, you can use the dart:html browser-only [`EventSource`](https://api.dart.dev/be/169657/dart-html/EventSource-class.html) but you'll have to handle the subscriptions and event messages parsing on your own.
